topydo/ui/prompt/PromptCompleter.py

```diff
@@ -0,0 +1,69 @@
+# Topydo - A todo.txt client written in Python.
+# Copyright (C) 2015 Bram Schoenmakers <bram@topydo.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This module provides a completer class that can be used by the prompt provided
+by the prompt toolkit.
+"""
+
+import re
+
+from prompt_toolkit.completion import Completer, Completion
+from topydo.ui.CompleterBase import CompleterBase, date_suggestions
+from topydo.lib.Config import config
+from topydo.lib.RelativeDate import relative_date_to_date
+
+
+def _dates(p_word_before_cursor):
+    """ Generator for date completion. """
+    to_absolute = lambda s: relative_date_to_date(s).isoformat()
+
+    start_value_pos = p_word_before_cursor.find(':') + 1
+    value = p_word_before_cursor[start_value_pos:]
+
+    for reldate in date_suggestions():
+        if not reldate.startswith(value):
+            continue
+
+        yield Completion(reldate, -len(value), display_meta=to_absolute(reldate))
+
+
+class PromptCompleter(Completer):
```

Comment:
We could also use multiple inheritance on `Completer` and `CompletionBase`, so we can get rid of `inner_completer`.